src/v/datalake/translation/partition_translator.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "base/outcome.h"
+#include "base/seastarx.h"
+#include "cluster/fwd.h"
+#include "datalake/errors.h"
+#include "datalake/fwd.h"
+#include "features/fwd.h"
+#include "model/record_batch_reader.h"
+#include "random/simple_time_jitter.h"
+#include "ssx/semaphore.h"
+#include "utils/prefix_logger.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/semaphore.hh>
+#include <seastar/core/shared_ptr.hh>
+
+namespace kafka {
+class read_committed_reader;
+}
+
+namespace datalake::translation {
+
+/**
+ * A partition translator is responsible for translating data from a given
+ * partition periodically. A single instance of translator should be active for
+ * a given partition and is attached to the leader replica.
```

Comment:
Should we callout that it's possible during leadership transitions for there to be two translators?